tests/python_compiler/test_quantum_dialect.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit test module for pennylane.compiler.python_compiler.quantum_dialect.py."""
+
+import pytest
+from xdsl.dialects.builtin import IntegerType
+from xdsl.dialects.test import TestOp
+from xdsl.ir import AttributeCovT, OpResult
+
+from pennylane.compiler.python_compiler.quantum_dialect import AllocOp, QuantumDialect, QuregType
```

Comment:
Using the following `pytest.mark.external` marker, these tests will run using the `external-libraries-tests` runner in CI.

```suggestion
from pennylane.compiler.python_compiler.quantum_dialect import AllocOp, QuantumDialect, QuregType

pytestmark = pytest.mark.external
```